node/impl/storminer.go

```diff
@@ -703,4 +705,15 @@ func (sm *StorageMinerAPI) ComputeProof(ctx context.Context, ssi []builtin.Secto
 	return sm.Epp.ComputeProof(ctx, ssi, rand)
 }
 
+func (sm *StorageMinerAPI) GetEnabledSubsystems(context.Context) (res []api.MinerSubsystem, err error) {
+	index := 1
+	for i := 0; i < api.MinerSubsystems; i++ {
+		if sm.Subsystems.Has(api.MinerSubsystem(index)) {
+			res = append(res, api.MinerSubsystem(index))
+		}
+		index *= 2
+	}
+	return
+}
+
```

Comment:
This seems very unnecessary. This should just be a:

```
type MinerSubsystems []string
```

And we should just return a []string.

Reply:
I think it is nicer to be able to just have this API `typed` when we use it in `lotus-miner info` for example.